### DISCUSSION TITLE
what happen when use use() in client side?

### DISCUSSION THREAD

**Discussion by premier213** [vercel/next.js#42243] [App Router]
I read the beta document next.js when creating a file with this code made a loop of requests to the server

[link to beta document example](https://beta.nextjs.org/docs/data-fetching/fetching#example-fetch-and-use-in-client-components)

```
"use client";

import { use } from "react";

async function getData() {
  const res = await fetch("https://api.example.com/...");
  // The return value is *not* serialized
  // You can return Date, Map, Set, etc.
  return res.json();
}

export default function Response() {
  const name = use(getData());
  console.log("🚀 ~ file: response.tsx ~ line 14 ~ Response ~ name", name);

  return "...";
}

```

![image](https://user-images.githubusercontent.com/4321174/199046380-d3be99aa-22cd-4616-bdc1-49475804c0f3.png)

**Comment by icyJoseph** [comment]
I think the caching functionality is not in place just yet. 

> Fetch deduping has been implemented in server components but the client component side hasn't been fully implemented yet. However in your case the fetch should really be done in a server component: DashboardClient has use client but it doesn't have to be a client component based on what was posted.

More info is discusses on this entry: https://github.com/vercel/next.js/discussions/41745#discussioncomment-3977857

Observe that React's still at 18.2.0, 5 months old as of writing, so nothing new is to be expected from their end on this Next 13 release.
